yarn-project/p2p/src/services/discv5/discV5_service.ts

```diff
@@ -196,22 +197,32 @@ export class DiscV5Service extends EventEmitter implements PeerDiscoveryService
     this.onDiscovered(enr);
   }
 
+  private isOurBootnode(enr: ENR) {
+    return this.bootstrapNodeEnrs.some(x => x.nodeId === enr.nodeId);
+  }
+
   private onDiscovered(enr: ENR) {
+    // Find out if this is one of our bootnodes
+    if (this.isOurBootnode(enr)) {
+      // it is, what we do here depends
+      if (!this.config.bootstrapNodesAsFullPeers) {
+        // we don't consider bootnodes as full peers, don't perform any checks and don't emit anything
+        return;
+      }
+      if (!this.config.bootstrapNodeEnrVersionCheck) {
+        // we do consider bootnodes to be full peers and we have been told to NOT version check them, so emit
+        this.logger.trace(`Skipping version check for bootnode ${enr.nodeId}`);
+        this.emit(PeerEvent.DISCOVERED, enr);
+        return;
+      }
+      // here, we do consider bootnodes as full peers and we must version check so we continue to regular validation
+    }
     if (this.validateEnr(enr)) {
       this.emit(PeerEvent.DISCOVERED, enr);
     }
   }
 
   private validateEnr(enr: ENR): boolean {
-    // Check if the peer is actually a bootnode and we have disabled the version check
-    if (
-      !this.config.bootstrapNodeEnrVersionCheck &&
-      this.bootstrapNodes.some(enrTxt => ENR.decodeTxt(enrTxt).nodeId === enr.nodeId)
-    ) {
-      this.logger.trace(`Skipping version check for bootnode ${enr.nodeId}`);
-      return true;
-    }
```

Comment:
Heads up `validateEnr` gets called in L95 as well. You'll have to place the bootnode check there as well if removed from here.

Reply:
I think I've got the logic correct now.